Readme.md

```diff
@@ -0,0 +1,243 @@
+sanitization of the report files.
+This means converting legacy YAML reports to JSON as well as converting all
+measurements to a consistent JSON format (and performing some fixes to the data
+format to avoid surprising consumers of the data).
+
+In other words, it converts tar.lz4 files in a 1:1 way.
+lz4 compression is a bit tricky at this stage: lz4 frames are independent,
+report offsets within the block are recorded in the separate index file, so the
+resulting file is both streamable (with ability to skip tar records without
+decoding), seekable (single lz4 frame is read to get a single report — that's
+~56kb of compressed data) and readable with ordinary `tar -I lz4 --extract` for
+whoever wants to parse sanitized measurements.
+
+Indexes for the reports are stored as well, they include block offsets, sizes,
+and offsets within block required to seek() to exact measurement and to read the
+file in streaming way without parsing tarfile stream.
+
+### Centrifugation
+
+**Centrifugation** is where the interesting things start happening. It's what is
+done to "enrich" the data.
+It reads data sanitized by autoclaving, extracts the essence that can be aggregated and
+stores metadata into a PostgreSQL database for further processing.
+
+Note: in the airflow DAG view this is actually called `meta_pg`.
+
+## Raw reports sensor
+
+This is not technically a task, but is rather a "sensor" in airflow lingo. What
+this means is that a check is done at `chameleon` (the host that aggregates all
+measurements from OONI collectors) to see if there is more work to do and if so
+will start a new iteration of the DAG.
+
+## Stack
+
+The technological stack of the pipeline is the following:
+
+* [Airflow](https://airflow.incubator.apache.org/) as a workflow manager or put
+ otherwise a "enhanced cron"
+* [PostgreSQL](https://www.postgresql.org/) as the primary database used to
+ store metadata
+* [Docker](https://www.docker.com/) to create containers for the various stages
+ of the pipeline, without having to worry about dependency management
+* [Ansible](https://www.ansible.com/) to make deployment automation easier
+
+
+## Usage
+
+In this section we will go over some of the common tasks related to operating
+and contributing to the OONI Pipeline.
+
+It's important to keep in mind that every task inside of the Airflow DAG needs
+to be optimized for big units of computation, you should avoid having small
+tasks, otherwise you incur in a lot of scheduling/orchestration overhead.
+
+### Modifying the centrifugation stage
+
+Most of the time you will probably care to edit the **centrifugation** stage
+instead of adding a new DAG task, to avoid re-reading the measurements multiple
+times.
+
+1. Edit the centrifugation file
+
+The file to edit is located in `af/shovel/centrifugation.py`.
+
+2. Push a new docker image
+
+You should then build a new docker image by bumping the version number inside of
+`af/shovel/build`.
+
+3. Redeploy an update
+
```

Comment:
Version number in `ansible/roles/airflow/files/docker-trampoline` should be bumped as well.